src/Data/NonEmpty.purs

```diff
@@ -34,33 +34,99 @@ import Data.Unfoldable1 (class Unfoldable1)
 -- | For example:
 -- |
 -- | ```purescript
+-- | import Data.NonEmpty
+-- |
+-- | nonEmptyArray :: NonEmpty Array Int
+-- | nonEmptyArray = NonEmpty 1 [2,3]
+-- |
+-- | import Data.List(List(..), (:))
```

Comment:
It might be preferable to move the import declarations to the top, as you aren't allowed to have any import declarations after the first non-import declaration in a module.

Reply:
I was trying to match the style of using the repl, and keeping things logically clustered. I guess it's a question of whether this slight addition of clarity/organization outweighs the confusion if someone pastes this into a .purs file. I think the error message is pretty clear though in that case.

Luckily this won't be a concern with doctest (repl style).

Reply:
Is this a blocker? I'm fine with changing it, but it doesn't seem too consequential either way, so I'll spare the effort if it's unnecessary.

Reply:
No, it’s just that changes to the core libraries should generally obtain two core team approvals, see https://github.com/purescript/governance